### PR TITLE
Adding feature_retrieval component

### PR DESCRIPTION
**Pull request by shail2208**

### CODE REVIEW

File: featurestore/components/feature_retrieval/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+display_name: Feature Retrieval
+description: Retrieval component to be used to retrieve features from feature store.
+is_deterministic: true
+
+code: ../src
+entry:
+  file: retrieval.py
+
+inputs:
```
Comment (elesel) [inline featurestore/components/feature_retrieval/spec.yaml:15]: Would you please add descriptions to inputs and outputs?

**Review by elesel — CHANGES_REQUESTED** [review]

File: featurestore/components/feature_retrieval/spec.yaml (hunk 1 above)
Comment (shail2208) [reply featurestore/components/feature_retrieval/spec.yaml:15]: Added descriptions

**Review by elesel — DISMISSED** [review]

**Review by elesel — DISMISSED** [review]

File: featurestore/components/feature_retrieval/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+$schema: http://azureml/sdk-2-0/SparkComponent.json
+
+name: feature_retrieval
+version: 0.1.0
+type: spark
+
+display_name: Feature Retrieval
+description: Retrieval component to be used to retrieve features from feature store.
```
Comment (feng-pan8014) [inline featurestore/components/feature_retrieval/spec.yaml:8]: retrieve offline features

File: featurestore/components/feature_retrieval/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      - parquet
+      - csv
+      - delta
+    description: Format of the observation data.
+
+outputs:
+  output_data:
+    type: uri_folder
+    mode: direct
+    description: Observation data with retrieved features.
```
Comment (feng-pan8014) [inline featurestore/components/feature_retrieval/spec.yaml:49]: Observation data with retrieved features written in parquet format

File: featurestore/components/src/retrieval.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import logging
+import os
+
+from azureml.dataprep.api._rslex_executor import ensure_rslex_environment
+from azureml.dataprep.rslex import Copier, PyIfDestinationExists, PyLocationInfo
+from pyspark.context import SparkContext
+from pyspark.sql.session import SparkSession
+
+from azureml.featurestore import FeatureStoreClient, get_offline_features
+from azureml.featurestore._identity import AzureMLHoboSparkOnBehalfOfCredential
```
Comment (feng-pan8014) [inline featurestore/components/src/retrieval.py:16]: use this

from azure.ai.ml.identity import AzureMLOnBehalfOfCredential

**Review by elesel — APPROVED** [review]

**Pull request merged**